raster/r.slope.aspect/main.c

```diff
@@ -541,427 +568,515 @@ int main(int argc, char *argv[])
         && dx_fd < 0 && dy_fd < 0 && dxx_fd < 0 && dyy_fd < 0 && dxy_fd < 0)
         exit(EXIT_FAILURE);
 
-    Rast_get_d_row_nomask(elevation_fd, elev_cell[2] + 1, 0);
-    if (Wrap) {
-        elev_cell[2][0] = elev_cell[1][Rast_window_cols()];
-        elev_cell[2][Rast_window_cols() + 1] = elev_cell[2][1];
-    }
-
     G_verbose_message(_("Percent complete..."));
 
-    for (row = 0; row < nrows; row++) {
-        /*  if projection is Lat/Lon, recalculate  V and H   */
-        if (G_projection() == PROJECTION_LL) {
-            north = Rast_row_to_northing((row - 1 + 0.5), &window);
-            ns_med = Rast_row_to_northing((row + 0.5), &window);
-            south = Rast_row_to_northing((row + 1 + 0.5), &window);
-            east = Rast_col_to_easting(2.5, &window);
-            west = Rast_col_to_easting(0.5, &window);
-            V = G_distance(east, north, east, south) * 4 / (factor * zfactor);
-            H = G_distance(east, ns_med, west,
-                           ns_med) * 4 / (factor * zfactor);
-            /* ____________________________
-               |c1      |c2      |c3      |
-               |        |        |        |
-               |        |  north |        |        
-               |        |        |        |
-               |________|________|________|          
-               |c4      |c5      |c6      |
-               |        |        |        |
-               |  west  | ns_med |  east  |
-               |        |        |        |
-               |________|________|________|
-               |c7      |c8      |c9      |
-               |        |        |        |
-               |        |  south |        |
-               |        |        |        |
-               |________|________|________|
-             */
-        }
-
-        G_percent(row, nrows, 2);
-        temp = elev_cell[0];
-        elev_cell[0] = elev_cell[1];
-        elev_cell[1] = elev_cell[2];
-        elev_cell[2] = temp;
-
-        if (row < nrows - 1)
-            Rast_get_d_row_nomask(elevation_fd, elev_cell[2] + 1, row + 1);
-        else
-            Rast_set_d_null_value(elev_cell[2], ncols + 2);
-
-        if (Wrap) {
-            elev_cell[2][0] = elev_cell[2][Rast_window_cols()];
-            elev_cell[2][Rast_window_cols() + 1] = elev_cell[2][1];
-        }
-
-        pc1 = elev_cell[0];
-        pc2 = elev_cell[1];
-        pc3 = elev_cell[2];
-
-        if (aspect_fd >= 0) {
-            asp_ptr = asp_raster;
-        }
-        if (slope_fd >= 0) {
-            slp_ptr = slp_raster;
-        }
-
-        if (pcurv_fd >= 0) {
-            pcurv_ptr = pcurv_raster;
-        }
+    int computed = 0; /* for computing progress */
+    int written = 0;
 
```

Comment:
`row = 0;` to avoid
```
main.c:1076:5: warning: ‘row’ may be used uninitialized in this function [-Wmaybe-uninitialized]
     G_percent(row, nrows, 2);
```

Reply:
Opps yeap it is supposed to be `G_percent(nrows, nrows, 2);`. Fixed it.